### PR TITLE
fix: add region argument to glue extractor boto3 client instantiation

### PR DESCRIPTION
**Pull request by AykCanDem**
### Summary of Changes

**glue extractor** does not get `region_name` argument. It might be needed in different scenarios. For example, while using **Databricks** for extracting data from AWS Glue, it gives `NoRegionError: You must specify a region.`

You might be questioning why cant we use aws configuration and set region with **.aws/config** $HOME directory. Databricks is using instance profile for associating cluster with AWS IAM role. It is not a good practice to install aws cli to databricks cluster and configure aws credentials. So, typically, there is no **.aws/config** file stored in databricks cluster, and it is not needed.

In a nutshell, it might be better to have an option to provide `region_name` parameter while creating glue extractor.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Review by mgorsk1 — COMMENTED** [review]
to be consistent across all keys I would do `REGION_NAME_KEY`: `region_name`. WDYT?

File: databuilder/databuilder/extractor/glue_extractor.py [hunk 1]
```diff
@@ -21,11 +21,13 @@ class GlueExtractor(Extractor):
...
     MAX_RESULTS_KEY = 'max_results'
     RESOURCE_SHARE_TYPE = 'resource_share_type'
+    REGION_NAME = "region"
     DEFAULT_CONFIG = ConfigFactory.from_dict({
         CLUSTER_KEY: 'gold',
         FILTER_KEY: None,
         MAX_RESULTS_KEY: 500,
-        RESOURCE_SHARE_TYPE: "ALL"
+        RESOURCE_SHARE_TYPE: "ALL",
+        REGION_NAME: None
```
Comment (mgorsk1) [inline databuilder/databuilder/extractor/glue_extractor.py:30]: If there is no default we don't really need this in the config

Comment (AykCanDem) [reply databuilder/databuilder/extractor/glue_extractor.py]: I agree about `REGION_NAME_KEY`. However, if we wont put it as default, tests gives errors `pyhocon.exceptions.ConfigMissingException: 'No configuration setting found for key region' `.
I think it should be there as `FILTER_KEY`.

**Comment by AykCanDem** [comment]
any update on this?

**Comment by mgorsk1** [comment]
closed in favor of https://github.com/amundsen-io/amundsen/pull/1687

**Pull request closed without merging**